libgalois/include/galois/worklists/AdaptiveObim.h

```diff
@@ -40,6 +39,32 @@
 namespace galois {
 namespace worklists {
 
+namespace internal {
+
+template <typename Index, bool UseDescending>
+struct AdaptiveOrderedByIntegerMetricComparator {
```

Comment:
When porting the code to the latest version I actually set things up to remove any reliance on the OBIM-style internal comparator classes since they really only exist to manage the direction the comparisons run. It's not a huge deal one way or the other, but I'd ultimately like to move OBIM and Adaptive OBIM away from using these helper classes since they're very verbose for what they actually are doing.

Reply:
That makes sense. Again, I was matching the code structure with OBIM assuming that OBIM's code was in a good shape. Now OBIM should be the one to change.

Reply:
Yah, these are really easy to change. I can just take care of it in a separate PR after we get this one sorted out.